File: javascript/leetcode/242. valid anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
  if (s.length !== t.length) return false;

  const mp = new Map();
  for (let val of s) {
    if (mp.has(val)) {
      mp.set(val, mp.get(val) + 1);
      continue;
    }
    mp.set(val, 1);
  }

  for (let val of t) {
    if (mp.has(val)) {
      mp.set(val, mp.get(val) - 1);
      continue;
    }
    return false;
  }

  for (let val of t) {
    if (mp.get(val) !== 0) return false;
  }

  return true;
};

File: go/leetcode/242. valid anagram.go
func isAnagram(s string, t string) bool { 
    map1 := make(map[int32] int) 
    
    if len(s) != len(t) {
        return false
    }
    
    for _, val := range s { 
        _, ok := map1[val]
        if ok {
            map1[val] = map1[val] + 1
            continue
        }
        map1[val] = 1
    }
    
    for _, val := range t {
        _, ok := map1[val]
        if ok {
            map1[val] = map1[val] - 1
            continue
        }  
        return false
    }
    
    for _, val := range t {
        if map1[val] != 0 {
            return false
        } 
    }
     
    return true
} 